source/app/requesthandlers/scheduler_setup_handler.py

```diff
@@ -222,22 +270,52 @@ def handle_request(self):
         finally:
             self._logger.flush()
 
-    def _update_settings(self):
+    def is_valid_org_id(self, org_id):
+        """
+        Verifies if the ou_id param is a valid ou_id format. https://docs.aws.amazon.com/organizations/latest/APIReference/API_Organization.html
+        :return: boolean
+        """
+        try:
+            return re.fullmatch("^o-[a-z0-9]{10,32}$", org_id)
+        except Exception as error:
+            raise error
+
+    def _update_settings(self, prev_org_remote_account_ids={}):
         try:
             admin = ConfigAdmin(logger=self._logger, context=self.context)
+            try:
+                org_id = list(self.remote_account_ids)[0]
+            except Exception as error:
+                self._logger.info(f"org id is not valid or empty {error}")
+                org_id = ""
+
+            if self.is_valid_org_id(org_id) and self.use_aws_organizations == "True":
+                self.organization_id = org_id
+                remote_account_ids = prev_org_remote_account_ids
+            elif self.is_valid_org_id(org_id) and self.use_aws_organizations == "False":
+                self.organization_id = org_id
+                remote_account_ids = {}
```

Comment:
This section seems to check for the use aws organizations param being equal to "True" or "False" but the options in cdk are "Yes" and "No".

A way to avoid problems like this is to decouple the layer that understands the meaning of particular properties from the event object and translates them into Python-friendly objects for the rest of the code to use. Taking the whole event/context as a param means that every part of this class must understand exactly what the event looks like.

Reply:
@gockle this is the only thing I consider a blocking issue, I'll leave the rest up to your discretion